sql/plan/show_create_table.go

```diff
@@ -73,7 +73,7 @@ func (sc ShowCreateTable) WithChildren(children ...sql.Node) (sql.Node, error) {
 	child := children[0]
 
 	switch child.(type) {
-	case *SubqueryAlias, *ResolvedTable, *UnresolvedTable:
+	case *SubqueryAlias, *ResolvedTable, *UnresolvedTable, *DeferredAsOfTable:
```

Comment:
we have `sql.UnresolvedTable`, which includes `*UnresolvedTable, *DeferredAsOfTable`, not sure which is preferable in this case but using the interface can help avoid missing switch conditions